### PR TITLE
GODRIVER-3096 Remove "stats" package dependency

### PR DESCRIPTION
**Pull request by timothy-kim-mongo**
GODRIVER-3096

## Summary

The changes in this pull-request answers the ticket GODRIVER-3096 of removing the "github.com/montanaflynn/stats" dependency from the Go driver module. This is done by moving the internal/benchmark code into internal/cmd/benchmark (the only placed where the "stats" package is used) to make the benchmark system its own submodule so it can pull in additional dependencies.

## Background & Motivation

[PR 1507](https://github.com/mongodb/mongo-go-driver/pull/1507) (tentatively) removes the need to calculate 90th percentile and standard deviation for RTT samples. Those are the last two "production" use cases for the "github.com/montanaflynn/stats" package in the Go driver. We should remove all other uses of the "stats" package so we can remove it from the Go driver dependencies.

### CODE REVIEW

**Comment by timothy-kim-mongo** [comment]
codespell failed in pre_commit test because of lorem ipsum string mispelled words in the test strings of internal/cmd/benchmark/operation_test.go

**Comment by blink1073** [comment]
> codespell failed in pre_commit test because of lorem ipsum string mispelled words in the test strings of internal/cmd/benchmark/operation_test.go

You can add that file to the exclude [regex](https://github.com/mongodb/mongo-go-driver/blob/4f7331dadbabdc820367b7eac4534894d7dfaa6a/.pre-commit-config.yaml#L44)

File: internal/cmd/benchmark/go.mod [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module go.mongodb.go/mongo-driver/internal/cmd/benchmark
+
+go 1.22
+
+toolchain go1.22.3
```
Comment (matthewdale) [inline internal/cmd/benchmark/go.mod:5]: Do we need to use Go 1.22 for anything specific? If not, I recommend removing the `toolchain` directive and specifying `go 1.20` to match the Go version we run in CI currently.

```suggestion
go 1.20
```

**Review by matthewdale — COMMENTED** [review]
One question/recommendation about Go versions, otherwise looks great!

File: internal/cmd/benchmark/go.mod (hunk 1 above)
Comment (timothy-kim-mongo) [reply internal/cmd/benchmark/go.mod]: I think this automatically updated from go mod tidy, but 1.22 isn't used for anything specific. Thank you!

**Review by blink1073 — APPROVED** [review]
LGTM!

**Pull request merged**